common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/trigger/DateTriggerConfiguration.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.module.trigger;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.Trigger;
+
+/**
+ * Configuration class for Date trigger.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+@Configuration
+@EnableConfigurationProperties(DateTriggerProperties.class)
+@ConditionalOnProperty("date")
+public class DateTriggerConfiguration {
+
+	@Autowired
+	private DateTriggerProperties dateTriggerProperties;
+
+	@Bean(name = TriggerConstants.TRIGGER_BEAN_NAME)
+	Trigger dateTrigger() {
+		Date date;
+		DateFormat dateFormat = new SimpleDateFormat(dateTriggerProperties.getDateFormat());
+		String dateString = dateTriggerProperties.getDate() != null ? dateTriggerProperties.getDate() :
+				dateFormat.format(new Date().toString());
```

Comment:
Looks like overhead: `new Date().toString()` to `parse` to `Date()` on the next line...